### PR TITLE
Add support for REPLACE statement on Mysql

### PR DESCRIPTION
**Pull request by hleon12**
Add support to REPLACE statement to Mysql flavour. It can be used like the `squel.insert()` function:

`squel = squel.useFlavour('mysql');`
`squel.replace().into(table).set(field, value)...`

In case that you're using other flavour, it throws an error.

More info: [http://dev.mysql.com/doc/refman/5.7/en/replace.html]

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -2111,6 +2116,9 @@ function _buildSquel(flavour = null) {
     insert: function(options, blocks) {
       return new cls.Insert(options, blocks);
     },
+    replace: function(options, blocks) {
```
Comment (hiddentao) [inline src/core.js]: Surely we can place this inside the `mysql` flavour and thereby get rid of the dummy class in `core`?

**Comment by hiddentao** [comment]
Thanks! Have commented.

File: src/core.js [hunk 2]
```diff
@@ -2091,12 +2091,7 @@ function _buildSquel(flavour = null) {
       super(options, blocks);
     }
   }
-
-  // Dummy REPLACE query builder
-  cls.Replace = function(options, blocks){
-    throw new Error(`Only available on Mysql`);
-  }
-
```
Comment (hleon12) [inline src/core.js]: Removed. Check the latest commit.

**Comment by hiddentao** [comment]
Thanks!

**Comment by YikaJ** [comment]
that's no any example in document

**Pull request merged**